### DISCUSSION TITLE
Is it possible to remove this state border overlay?

### DISCUSSION THREAD

**Discussion by formula4** [ImageMagick/ImageMagick#6385] [General] [answered]
I have an image of the United States (satellite data) which includes an overlay of state outlines. I wish to remove them, but even after digging into ImageMagick documentation I am not sure what I could do or if it's possible. Here is a snip of the image, where the white outlines are clearly visible. (I am also including a link of the actual data I need changed, one is the state overlay itself without underlying imagery - so I am not sure if that could helpful). Both images are pixel perfect together.

Edit : I asked how to "remove" them, of course that could mean many things. I am open to any suggestions, if somehow each pixel could be filled in by the nearest neighbor, or blurred together, anything except the awful white lines.

Link to actual data : https://drive.google.com/file/d/1J2-f8i80EDdNs9W_cnjv7WU0APbgdc6U/view?usp=sharing (fixed link)

![image](https://github.com/ImageMagick/ImageMagick/assets/56749434/bcc05255-9f39-4f61-9e41-8ba6a6aa55c4)

Any ideas?

**Comment by fmw42** [comment]
Your link does not work for me.

**Reply by formula4** [reply]
https://drive.google.com/file/d/1J2-f8i80EDdNs9W_cnjv7WU0APbgdc6U/view?usp=sharing (I fixed the link in the main post, thank you)

**Reply by fmw42** [reply]
This link is OK. I downloaded your files.

**Comment by fmw42** [comment, marked as answer]
With Imagemagick, you can do a 7x7 median filter on the image. Then use the b/w map image as a mask to blend so that the median is used where the lines are and the rest of the image comes from the original.

Unix syntax for Imagemagick 7:

`magick -quiet channel01-prepped.tif \( +clone -statistic median 7x7 \) CONUS-EX-OUTLINES.tif -compose over -composite -compress LZW channel01-prepped_result.tif`


The result is too big to post.  But you can run the command above.

The alternate would be inpainting.  But Imagemagick does not include that.  

Please follow the instructions when posting and always supply your Imagemagick version and platform.

**Reply by formula4** [reply]
I was not expecting such a great result, it is much more muted down. So Impressive. - Many thanks, I will be looking further into this.

**Reply by fmw42** [reply]
You can fine tune the median filter size to optimize your result.

**Reply by formula4** [reply]
Thank you for the advice. i am already making tweaks. And perhaps better than a thank you - results. The image I gave was actually one of 3 images (R,G,B) to build a color satellite image. The result is looking remarkable without the borders in the way. I am convinced ImageMagick really is magic.
![image](https://github.com/ImageMagick/ImageMagick/assets/56749434/75eab4c6-784d-4918-84b0-f66801bc99a0)

**Reply by fmw42** [reply]
What was your new improved command line?

**Reply by formula4** [reply]
I played with the numbers and it seems 7x7 is the magic spot. I am not good enough with ImageMagick yet to test other functions on it. My briliant idea was to actually run your script twice. Which, it had a very positive effect. Running it 3x, however, seemed to have no effect. So the best I have as of now is 2x as in the screenshot, which is very satisfactory. I would be interested (out of curiosity) if there other ideas to take it beyond this. 

Edit : The picture I sent above looks cleaner because it is so zoomed out, so again this is a great result!

![image](https://github.com/ImageMagick/ImageMagick/assets/56749434/aba3fc5f-2239-4763-98f3-22824bbfad54)

![image](https://github.com/ImageMagick/ImageMagick/assets/56749434/77363394-35a9-447e-8237-0e6b055a23e7)
